File: final_front_end_angular/src/app/body/api/consultation/Eyelid.ts
export interface Eyelid {
    id: number;
    createdBy: string;
    createdAt: string;
    modifiedBy: string;
    modifiedAt: string;
    deleted: boolean;
    version: number;

    eyeSide: 'OU' | 'OD' | 'OS' | null;
    normal: boolean;
    entropion: boolean;
    ectropion: boolean;
    ptosis: boolean;
    anteriorBlepharitis: boolean;
    staphylococcus: boolean;
    demodex: boolean;
    other: boolean;
    otherText?: string | null;
}

File: final_back_end_spring_boot/src/main/java/com/jorami/eyeapp/model/consultation/Eyelid.java
package com.jorami.eyeapp.model.consultation;

import com.jorami.eyeapp.model.Enum;
import com.jorami.eyeapp.model.IdentifiedModel;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.*;
import org.hibernate.annotations.Where;
import org.hibernate.envers.Audited;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Audited
@Where(clause = "deleted = FALSE")
@Entity
@Table(name="eyelid")
public class Eyelid extends IdentifiedModel {
    @Enumerated(EnumType.STRING)
    private Enum.EyeSide eyeSide;
    private boolean normal;
    private boolean entropion;
    private boolean ectropion;
    private boolean ptosis;
    private boolean anteriorBlepharitis;
    private boolean staphylococcus;
    private boolean demodex;
    private boolean other;
    private String otherText;
}
